execution/executor-service/src/remote_state_view.rs

```diff
@@ -114,25 +123,61 @@ impl RemoteStateViewClient {
 
     pub fn init_for_block(&self, state_keys: Vec<StateKey>) {
         *self.state_view.write().unwrap() = RemoteStateView::new();
-        self.pre_fetch_state_values(state_keys);
+        info!(
+            "REMOTE_EXECUTOR: Prefetching {} state keys",
+            state_keys.len()
+        );
+        self.pre_fetch_state_values(state_keys, false);
     }
 
-    fn pre_fetch_state_values(&self, state_keys: Vec<StateKey>) {
+    fn insert_keys_and_fetch_values(
+        state_view_clone: Arc<RwLock<RemoteStateView>>,
+        thread_pool: Arc<ThreadPool>,
+        kv_tx: Arc<Sender<Message>>,
+        shard_id: ShardId,
+        state_keys: Vec<StateKey>,
+    ) {
         state_keys.clone().into_iter().for_each(|state_key| {
-            self.state_view.read().unwrap().insert_state_key(state_key);
+            state_view_clone.read().unwrap().insert_state_key(state_key);
         });
         state_keys
             .chunks(REMOTE_STATE_KEY_BATCH_SIZE)
             .map(|state_keys_chunk| state_keys_chunk.to_vec())
             .for_each(|state_keys| {
-                let sender = self.kv_tx.clone();
-                let shard_id = self.shard_id;
-                self.thread_pool.spawn(move || {
+                let sender = kv_tx.clone();
+                thread_pool.clone().spawn(move || {
                     Self::send_state_value_request(shard_id, sender, state_keys);
                 });
             });
     }
 
+    fn pre_fetch_state_values(&self, state_keys: Vec<StateKey>, sync_fetch: bool) {
```

Comment:
Please add a comment explaining why `sync_fetch` is needed.

Reply:
Done